### ISSUE TITLE
Reagent bag shows wrong tooltips

### ISSUE THREAD

**Issue by cremor** [gsnerf/Baud-Bag#6]
The reagent bag content slots show wrong tooltips. Doesn't matter if the slots are empty or not, they always show tooltips from items in the bank.

**Comment by cremor** [comment]
I found something weird: The reagent bag tooltips are correct while the bank is open. But as soon as you close the bank, the tooltips are wrong again.

**Comment by Ketrel** [comment]
I'm having the same issue as of 10.0.2.8

![Wow_2022-11-27_22-16-45](https://user-images.githubusercontent.com/1002607/204183943-c39006ce-137f-4b19-a5f1-99c563465800.png)

![Wow_2022-11-27_22-16-25](https://user-images.githubusercontent.com/1002607/204183950-4d86eecd-400b-454a-849e-1f4c1b7cc479.png)

![Wow_2022-11-27_22-17-55](https://user-images.githubusercontent.com/1002607/204184066-f44eb78b-e046-4138-88b8-53825c6a8dcb.png)

Same item stack in and out of the reagent bag.  Also depending on which slot you mouseover the item from, you get odd borders too.  I've also gotten the corrupted item border from BFA.

**Comment by Ketrel** [comment]
Addendum: this appears to affect bag slots: 5, 6, 7, 14, 15, 17, 19, 20 in the reagent bag for me.  I'm not sure if that's consistent for everyone.

**Comment by gsnerf** [comment]
Sorry for the delay... somehow I **still** don't get notifications for the issues created here...

This seems to be an issue with the item link cache. Will analyse and fix shortly.

**Comment by gsnerf** [comment]
The issue is again with the changed ID structuring for bugs.

TL;DR: fixed with the next version.

Explanation:
With 10.0.2 IDs for Reagent Bags have been added between backpack bags (0-4) and bank bags (previously 5-11, now 6 to 12). So the cache still holds values for bag 5 (first bank bag) and the caching mechanism didn't yet know about the new ID system, so it didn't stop retrieval from cached values (and also didn't clean it up). So creating the tooltips for the items in the bag always retrieved the old cached values (that possibly don't even match the current bank bag content).

Fixed the caching mechanism to know about the new ID structure and also adapted it to use the constants introduced in the last few expansions to harden against future changes.

**Comment by Ketrel** [comment]
1074e3a does appear to correct it on my end too

**Comment by cremor** [comment]
Fixed in 10.0.2.9, thanks!
